### PR TITLE
Enable and fix Qt6 build CI test

### PR DESCRIPTION
**Pull request by nirvn**
This PR enables the Qt6 build CI test against master :tada: 

It allows us to lock in the recent improvements to insure core, authentication, and providers can compile against qt6 and avoid regressions there.

The next step should be to enable tests and witness the wave of failures :) I've tried to enable tests in this PR but ultimately left it out of this PR's set of goals due to qt6-specific compilation errors that need addressing and having run out of free time to do this.

### CODE REVIEW

File: src/core/symbology/qgssymbollayer.h [hunk 1]
```diff
@@ -40,7 +41,6 @@ class QgsDxfExport;
 class QgsExpression;
 class QgsRenderContext;
 class QgsPaintEffect;
-class QgsSymbolLayerReference;
```
Comment (nyalldawson) [inline src/core/symbology/qgssymbollayer.h:43]: Can we avoid this with an explicit destructor?

**Comment by nyalldawson** [comment]
Nice work @nirvn! Can you pull out the individual code fixes (not ci work) into atomic prs so we can review them individually?

File: src/core/symbology/qgssymbollayer.h (hunk 1 above)
Comment (nirvn) [reply src/core/symbology/qgssymbollayer.h:43]: @nyalldawson , I'm not sure, I'll try in the next run :)

Comment (nirvn) [reply src/core/symbology/qgssymbollayer.h]: @nyalldawson , nope, that's not enough.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**